web/rest/client/src/Controller/Provider/PostHolidaysMassImportAction.php

```diff
@@ -42,27 +45,42 @@ public function __invoke(Request $request)
         }
 
         $calendarId = $request->request->get('calendar');
+
+        /** @var CalendarInterface|null $calendar */
+        $calendar = $this->calendarRepository->find($calendarId);
+        if (is_null($calendar)) {
+            throw new NotFoundHttpException('Calendar not found');
+        }
+
+        $calendarBelongsToCompany = $calendar->getCompany()->getId() === $company->getId();
```

Comment:
What about a new repository method findCalendarByCompanyId($companyId, $calendarId), not worth it?

Reply:
Or maybe findCompanyCalendar($companyId, $calendarId) following the nomenclature found here: https://github.com/irontec/ivozprovider/blob/6bcc2584e430b2773ac4d2030b927e6dd8828393/library/Ivoz/Provider/Domain/Model/Extension/ExtensionRepository.php#L19